include/chainbase/chainbase.hpp

```diff
@@ -774,6 +788,39 @@ namespace chainbase {
 
              idx_ptr->validate();
 
+            // Ensure the undo stack of added index is consistent with the other indices in the database
+            if( _index_list.size() > 0 ) {
+               auto expected_revision_range = _index_list.front()->undo_stack_revision_range();
```

Comment:
What happens if the first index added is a new one with no undo states: (-1 -1).  and the second index added is more representative of the state of other indices?  Would it be possible to process this after all indices are added so that we can enforce that all revisions are equal to the maximum revision found in all added indices?